### PR TITLE
build debug packages

### PR DESCRIPTION
**Pull request by cbix**
**Read**
- https://wiki.archlinux.org/title/Debugging/Getting_traces
- https://wiki.archlinux.org/title/Debuginfod
- https://sleepmap.de/2022/packaging-for-arch-linux/#debug-packages

**TODO**
- [x] fix detecting missing packages
- [x] publish pipeline
- [x] debuginfod setup
- [x] documentation

For now I'm only bumping two pkgrels but eventually all packages should get rebuilt with the debug option enabled.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,3 +29,15 @@ Download, import and sign the current signing key:
...
+
+## Debug packages
+[Similar to](https://wiki.archlinux.org/title/Debugging/Getting_traces) the official repos, we provide debug symbols in separate packages and
+run a Debuginfod server under `https://arch.osamc.de`. To use this with compatible clients like `gdb`, install the `debuginfod` package and
+add the URL:
+```
+# pacman -Sy debuginfod
+# echo https://arch.osamc.de > /etc/debuginfod/proaudio.urls
+```
+Reboot, so the `DEBUGINFOD_URLS` environment variable gets initialized correctly.
```
[the remark this answers is not in the document]
Comment (cbix) [reply README.md:41]: Re-login didn't work for me, but [the wiki](https://wiki.archlinux.org/title/Environment_variables#Globally) is also a bit unclear about `profile.d` variables...

File: .drone.yml [hunk 2]
```diff
@@ -262,18 +283,20 @@ steps:
     failure: ignore
     environment:
       PKGDEST: /drone/src/out
-      MAKEFLAGS: -j4
```
[the remark this answers is not in the document]
Comment (SpotlightKid) [reply .drone.yml:265]: Got it.

File: tools/prepare-build.sh [hunk 3]
```diff
@@ -98,7 +98,8 @@ if [[ -f "$ROOT"/out/proaudio.db.tar.gz ]]; then
     for PKG in $ALLPKGS; do
         IGN=true
         PKGFILES=($(cd $PKG; makepkg --packagelist))
-        for PKGPATH in "${PKGFILES[@]}"; do
+        # FIXME: only remove -debug when it is a suffix. A package name like "my-debug-machine" should be kept.
+        for PKGPATH in "${PKGFILES[@]/\-debug}"; do
```
[the remark this answers is not in the document]
Comment (SpotlightKid) [reply tools/prepare-build.sh:102]: Ok, this seems a general Arch issue with debug package file names. Let's cross that bridge when we get there :)

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (SpotlightKid) [reply README.md:41]: Well, better be safe than sorry, I guess.

**Pull request merged**